src/content/blog/first-four-val-town-runtimes.mdx

```diff
@@ -0,0 +1,247 @@
+---
+title: The first four Val Town runtimes
+generated: 1701894028894
+description: From vm to vm2 to workers to processes, the long journey to how we run vals today
+pubDate: February 5, 2024
+heroImage: ./first-four-val-town-runtimes/timeline.png
+author: Tom MacWright
+---
+
+import { Image } from "astro:assets";
+import memoryImage from "./first-four-val-town-runtimes/memory.png";
+
+One of the most important things that Val Town does is **run untrusted code**.
+Our whole premise requires us to take the TypeScript code that our community
+writes and run it on a server, safely, quickly, and flexibly.
```

Comment:
```suggestion
We take your TypeScript code and run it on a server, safely, quickly, and flexibly.
```

Reply:
You could call out that there are tradeoffs between all of these items, and figuring out how to balance them all, and get the most of each is the trick, and we keep iterating in that direction, blah blah blah (sorry, writing quick notes!)